src/main.rs

```diff
@@ -120,24 +120,38 @@ fn main() {
                     }
                     Message::HotkeyPressed(hotkey_type) => unsafe {
                         if hotkey_type == HotkeyType::Maximize {
-                            if let Some(mut window) = Some(Window(GetForegroundWindow())) {
-                                let mut grid = GRID.lock().unwrap();
-                                let max_rect = grid.get_max_area();
-                                let window_rect = window.rect();
-
-                                if let Some((_, previous_rect)) = grid.previous_resize {
-                                    if window_rect == max_rect {
-                                        grid.previous_resize = Some((window, window_rect));
-                                        window.set_pos(previous_rect, None);
-                                    } else {
-                                        grid.previous_resize = Some((window, window_rect));
-                                        window.set_pos(max_rect, None);
-                                    }
+                            let mut grid = GRID.lock().unwrap();
+
+                            let mut active_window = if grid_window.is_some() {
+                                grid.active_window.unwrap()
```

Comment:
Why do you need to check `grid_window` to get `grid.active_window`?

Reply:
Because if the grid is open and we call `GetForegroundWindow()`, it could select and maximize the grid, which wouldn't be good. When the grid is open, any active window changes are synced to `grid.active_window`, so instead we just take it from there.

Reply:
There are a lot of redundancies throughout the program in updating `GRID`... it was easily to be lazy about it after making it globally mutable, yay lazy_static + Arc<Mutex<T>> :P But now that we've added so many additional features, it's definitely a pain point and there are a lot of potential potholes to track when making changes.